packages/react-native-web/src/exports/Modal/ModalFocusTrap.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import React, { useRef, useCallback, useEffect } from 'react';
+
+import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
+
+import View from '../View';
+import createElement from '../createElement';
+import StyleSheet from '../StyleSheet';
+
+/**
+ * This Component is used to "wrap" the modal we're opening
+ * so that changing focus via tab will never leave the document.
+ *
+ * This allows us to properly trap the focus within a modal
+ * even if the modal is at the start or end of a document.
+ */
+
+const FocusBracket = () => {
+  return createElement(
+    'div',
+    {
+      style: styles.focusBracket,
+
+      // Sets `aria-hidden` to true & remove the semantic meaning
+      accessible: true,
+      importantForAccessibility: 'no-hide-descendants',
+      accessibilityRole: 'none',
+
+      // `importantForAccessibility` being set to `no-hide-descendants`  will prevent
+      // these two attributes from being set as needed.
+      'data-focusable': true,
+      tabIndex: 0
+    }
+  );
+};
+
+function attemptFocus(element: any) {
+  if (!canUseDOM) {
+    return false;
+  }
+
+  try {
+    element.focus();
+  } catch (e) {
+    // Do nothing
+  }
+
+  return document.activeElement === element;
+}
+
+function focusFirstDescendant(element: any) {
+  for (let i = 0; i < element.childNodes.length; i++) {
+    const child = element.childNodes[i];
+    if (attemptFocus(child) || focusFirstDescendant(child)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function focusLastDescendant(element: any) {
+  for (let i = element.childNodes.length - 1; i >= 0; i--) {
+    const child = element.childNodes[i];
+    if (attemptFocus(child) || focusLastDescendant(child)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+export type ModalFocusTrapProps = {|
+  children?: any,
+  active?: boolean | () => boolean,
+|};
+
+const ModalFocusTrap = ({ active, children }: ModalFocusTrapProps) => {
+  const trapElementRef = useRef<?Node>();
+
+  // Ref used to track trapping of focus and to prevent focus from leaving a modal
+  // for accessibility reasons per W3CAG.
+  const focusRef = useRef<{ trapFocusInProgress: boolean, lastFocusedElement: ?HTMLElement }>({
+    trapFocusInProgress: false,
+    lastFocusedElement: null
+  });
+
+  const trapFocus = useCallback(() => {
+    // We should not trap focus if:
+    // - The modal hasn't fully initialized with an HTMLElement ref
+    // - Focus is already in the process of being trapped (eg, we're refocusing)
+    // - isTrapActive prop being false-ish tells us to do nothing
+    // - we cannot use the DOM
+    if (!trapElementRef.current || focusRef.current.trapFocusInProgress || !active || !canUseDOM) {
```

Comment:
Shouldnt need canUseDOM here, because this function is only called from within a canUseDOM condition.